### PR TITLE
Bolt trigger

### PR DESCRIPTION
**Pull request by danielamorse**
## Jira

- http://vjira2:8080/browse/BDS-1147

## Summary

Add Bolt Trigger component.

## Details

Bolt Trigger provides the "actionable" properties of `bolt-button` or `bolt-link` without the default styles. Use it as an unstyled wrapper around an element you'd like to make actionable. We plan to use `bolt-trigger` on the `bolt-modal` close button and image zoom.

## How to test

Review all `bolt-trigger` [demos](https://feature-bolt-trigger.boltdesignsystem.com/pattern-lab/patterns/02-components-trigger/index.html).

~_Note: This PR is just to review the `bolt-trigger` component. More modal-specific work (e.g. setting up the close button) will be done in the modal branch after this work is merged._~

### CODE REVIEW

**Comment by mikemai2awesome** [comment]
Nice start, @danielamorse! Here are some initial thoughts:

1. Tag prop can be eliminated, it should be determined by whether a `url` is passed.
2. There should be a prop to turn off focus styles (ie. using the trigger to build a custom close button with its own unique focus styles).
3. The image trigger is still a bit hacky, it requires the user to not just wrap the image with a trigger, but also wrap the image with a container that has a defined width. 
    * `bolt-image` is using `width: 100%`, the `100%` here or any percentage value is not determinable when the parent container does not have a defined width (ie. if display is not set to block/flex, or if position is set to fixed/absolute without width, or if width is set to auto). 
    * This could be solved if `bolt-image` has control of its `width` and `max-width`, by default `width: 100%` and `max-width: true-size` where true size means the image file's true size (ie. 400px). What that means is the image will take up the full width of the parent container if that container is equal to or less than 400px, otherwise the image will not go above 400px. If those defaults are set, a `bolt-image` can be dropped anywhere without worrying about whether the parent container has a width or not.

**Comment by danielamorse** [comment]
> 1. Tag prop can be eliminated, it should be determined by whether a `url` is passed.
Done

> 2. There should be a prop to turn off focus styles
Done

> 3. The image trigger is still a bit hacky
Agreed. I think 'true-size' is the right approach. Let's address that in the modal branch (if we have time before MVP).

File: packages/components/bolt-image/src/image.js [hunk 1]
```diff
@@ -215,6 +217,14 @@ class BoltImage extends withLitHtml() {
       this.style.backgroundColor = placeholderColor;
     }
 
+    if (maxWidth) {
+      this.style.width = maxWidth;
+      // Don't bother setting `max-width` if `width` is also a percentage
```
Comment (sghoweri) [inline packages/components/bolt-image/src/image.js]: 👍

File: packages/components/bolt-image/src/image.twig [hunk 2]
```diff
@@ -45,6 +45,12 @@
 {# Only JPGs allowed, PNGs can have transparency and may not look right layered over placeholder #}
 {% set can_use_placeholder = can_use_ratio or cover %}
 
+{# Setup inline styles that will be added to the custom element  #}
```
Comment (sghoweri) [inline packages/components/bolt-image/src/image.twig]: @danielamorse why are we setting these inline styles this way?! 

We already have an attributes object -- why aren't we just setting our `style` key on the attributes object? I haven't tested this out locally but I'm assuming we could probably just do something like this on line 52 and remove what you currently have on lines 52 and 130:

```
{% set attributes = attributes | merge({
  style: [
    inline_bg_color,
    inline_width,
    inline_max_width,
  ]
}) %}
```

File: packages/components/bolt-trigger/trigger.schema.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  cursor:
+    description: Type of cursor shown on hover.
+    type: string
+    default: pointer
+    enum:
+      - default
+      - pointer
+      - zoom-in
+      - zoom-out
+  on_click:
```
Comment (sghoweri) [inline packages/components/bolt-trigger/trigger.schema.yml]: Reminder: we'll likely be deprecating or removing this older declarative event syntax in v3.0 -- no sense worrying too much about it though. At least we have it documented 👍

File: packages/components/bolt-trigger/trigger.schema.yml [hunk 4]
```diff
@@ -0,0 +1,45 @@
+title: Bolt Trigger
+description: Triggers add button or link behavior to any content without the default button or link styles.
+type: object
+properties:
+  attributes:
+    type: object
+    description: A Drupal-style attributes object with extra attributes to append to this component.
+  content:
```
Comment (sghoweri) [inline packages/components/bolt-trigger/trigger.schema.yml]: ^ Twig-only prop

File: packages/components/bolt-image/src/image.twig (hunk 2 above)
Comment (danielamorse) [reply packages/components/bolt-image/src/image.twig]: Good call. Fixed 👍

File: packages/components/bolt-trigger/trigger.schema.yml (hunk 4 above)
Comment (danielamorse) [reply packages/components/bolt-trigger/trigger.schema.yml]: Updated description.

**Pull request merged**